scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -3310,13 +3312,54 @@ static bool antistall_scan(void)
 	return true;
 }
 
+static void tick_layer(struct cpu_ctx *cpuc, u32 layer_id)
+{
+	u64 dsq_id = layer_dsq_id(layer_id, cpuc->llc_id);
+	if (!scx_bpf_dsq_nr_queued(SCX_DSQ_LOCAL_ON | cpuc->cpu) &&
+	    !scx_bpf_dsq_nr_queued(dsq_id))
+		return;
+
+	scx_bpf_kick_cpu(cpuc->cpu, SCX_KICK_PREEMPT);
+}
+
+
+/*
+ * This function runs as a timer to implement tickless layer scheduling.
+ */
+bool tickless_timer(void)
+{
+	struct cpu_ctx *cpuc;
+	u64 layer_id;
+	int cpu;
+
+	if (!enable_tickless)
+		return false;
+
+	bpf_for(cpu, 0, nr_possible_cpus) {
+		if (!(cpuc = lookup_cpu_ctx(cpu))) {
+			scx_bpf_error("failed to lookup cpu_ctx");
+			return 0;
+		}
+
+		bpf_for(layer_id, 0, nr_layers) {
+			if (cpuc->layer_id != layer_id)
```

Comment:
`cpu_ctx` also has a `task_layer_id` field as well, but I think this is right because it's the cpu "owned" layer